charters/ingestion-service/charter.md

```diff
@@ -0,0 +1,177 @@
+  * Upload metadata describing biological materials, data files, and the experimental processes that generated them
+  * Validate metadata against endorsed community standards
+  * Interact with the DCP Upload Service to track and validate uploaded data files
+  * Track submission states that describe the life-cycle of experimental data as it moves through the DCP
+  * Generate identifiers for data, metadata and submissions
+  * Store submitted data in the DCP Data Storage Service
+
+### Definitions
+
+*Data contributor* - any user who uploads data into the DCP
+
+*Data consumer* - any user who accesses or downloads data from the DCP
+
+*Direct data contributor* - a data contributor who uploads their own data through services offered as part of the DCP in a "self-service" manner. Direct data contributors are also usually the generators of the data
+
+*Data wrangler* - an individual who works on behalf of the Human Cell Atlas (HCA) consortium to provide assistance to data contributors uploading data into the DCP. From an Ingestion Service perspective, data wranglers can themselves be considered as a specialised type of data contributor
+
+*Submission* - the grouping of data and metadata that makes sense for a data contributor to supply as part of a single interaction with the Ingestion Service. For example, a data contributor might typically upload all the data at once for a publication they have in preparation. This grouping does not necessarily correspond to how data is stored or displayed in the DCP and how it should be displayed to a data consumer
+
+*Bundle* - the grouping of data and metadata in the Data Storage Service
+
+## Objectives
+
+The Ingestion Service supports data contributors in the collection, validation and submission of data and metadata at the point of generation.
+
+The primary mission of the Ingestion Service is to define a publicly accessible application programming interface - the **Ingest API** - for the upload of data and metadata that will be stored in the Data Storage Service and that adheres to endorsed community standards.
+
+The secondary mission is to provide convenient, user-friendly methods for the upload of data and metadata to the DCP. The Ingestion Service will provide user interfaces and tools suitable for a variety of data contributor profiles, and will continuously review whether these interfaces and tools adequately serve the needs of the HCA user community. As part of this secondary mission, the Ingestion Service will deliver two main interfaces that utilise the Ingest API to support ingestion of data:
+
+  * A **User Interface** targeting data contributors uploading data and metadata via a form-based user interface
+  * A **Spreadsheet Upload** service targeting data generators uploading experimental metadata using spreadsheets in single, canonical defined spreadsheet format based on a specific Excel template (.xlsx) created by the spreadsheet template generator
```

Comment:
> based on a specific Excel template (.xlsx)

I'm worried about locking the Ingest Service in to a single spreadsheet implementation format here (although I appreciate that this is the only format the Ingest Service supports at the moment).

Reply:
Given that we make extensive use of tabs to represent different metadata sheets, the common alternatives (CSV, TSV) aren't appropriate. OpenDocument (.ods) format might be, but I think this will be more work than benefit (I've never seen a biologist save a spreadsheet in Excel using anything other than .xslx). Supporting Google Sheets seems like a very good idea, though. I think that covers the whole landscape though.

Summarising, I think the options are to to remove the "Excel" part, or add additional text, maybe something like

> "...based on a specific template available for Excel (.xlsx) or Google Sheets..."

Removing any reference seems a bit enigmatic given we're talking about a very small number of routes that will be supported. Would you prefer one over the other? Or something else?

Reply:
I mean, someone might be using old .xls Excel format :)

I'm happy to just have Excel and Google Sheets mentioned.